lib/srv/desktop/rdp/rdpclient/src/rdpdr/tests.rs

```diff
@@ -1197,3 +1198,65 @@ fn test_scard_ioctl_readcachew() {
         )],
     );
 }
+
+#[test]
+fn test_scard_ioctl_writecachew() {
+    let context_value = 5;
+    let mut c = client(true, context_value, Some(context_value - 1));
+
+    test_payload_in_to_response_out(
+        &mut c,
+        PayloadIn {
+            channel_pdu_header: ChannelPDUHeader {
+                length: 240,
+                flags: ChannelPDUFlags::CHANNEL_FLAG_FIRST
+                    | ChannelPDUFlags::CHANNEL_FLAG_LAST
+                    | ChannelPDUFlags::CHANNEL_FLAG_ONLY,
+            },
+            shared_header: SharedHeader {
+                component: Component::RDPDR_CTYP_CORE,
+                packet_id: PacketId::PAKID_CORE_DEVICE_IOREQUEST,
+            },
+            request: Box::new(DeviceControlRequest {
+                header: DeviceIoRequest {
+                    device_id: 1,
+                    file_id: 1,
+                    completion_id: 2,
+                    major_function: MajorFunction::IRP_MJ_DEVICE_CONTROL,
+                    minor_function: MinorFunction::IRP_MN_NONE,
+                },
+                output_buffer_length: 2048,
+                input_buffer_length: 184,
+                io_control_code: IoctlCode::SCARD_IOCTL_WRITECACHEW,
+            }),
+            scard_ctl: Some(Box::new(WriteCache_Call {
+                lookup_name: "Cached_CardProperty_Read Only Mode_0".to_string(),
+                common: WriteCache_Common {
+                    context: Context {
+                        length: 4,
+                        value: context_value,
+                    },
+                    card_uuid: vec![
+                        138, 113, 14, 35, 145, 213, 78, 249, 174, 208, 142, 171, 174, 121, 3, 76,
```

Comment:
I understand that the API is very low level, but are there any chances you could add some description of the data? I'd imagine that debugging this will be painful.

Reply:
I'm actually not very well informed about what goes on within the `scard` module, I was just copy-pasting this from the logs. I'll take a closer look.